packages/create-plugin/src/migrations/scripts/003-update-eslint-deprecation-rule.test.ts

```diff
@@ -0,0 +1,125 @@
+import { describe, expect, it } from 'vitest';
+import migrate from './003-update-eslint-deprecation-rule.js';
+import { Context } from '../context.js';
+
+describe('003-update-eslint-deprecation-rule', () => {
+  it('should not update ESLint config if no deprecation rule is present', () => {
+    const context = new Context('/virtual');
+
+    context.addFile(
+      '.config/.eslintrc',
+      JSON.stringify({
+        overrides: [
+          {
+            files: ['src/**/*.{ts,tsx}'],
+          },
+        ],
+      })
+    );
+
+    const initialChanges = context.listChanges();
+    migrate(context);
+    expect(context.listChanges()).toEqual(initialChanges);
+  });
+
+  it('should update ESLint config and package.json', () => {
+    const context = new Context('/virtual');
+    context.addFile(
+      '.config/.eslintrc',
+      JSON.stringify({
+        overrides: [
+          {
+            plugins: ['deprecation'],
+            files: ['src/**/*.{ts,tsx}'],
+            rules: {
+              'deprecation/deprecation': 'warn',
+            },
+          },
+        ],
+      })
+    );
+    context.addFile(
+      'package.json',
+      JSON.stringify({
+        devDependencies: {
+          'eslint-plugin-deprecation': '^2.0.0',
+          '@typescript-eslint/eslint-plugin': '^6.18.0',
+          '@typescript-eslint/parser': '^6.18.0',
+        },
+      })
+    );
+
+    const result = migrate(context);
+
+    // Check ESLint config changes
+    const eslintConfig = JSON.parse(result.getFile('.config/.eslintrc') || '{}');
+    expect(eslintConfig.overrides[0].plugins).toBeUndefined();
+    expect(eslintConfig.overrides[0].rules['deprecation/deprecation']).toBeUndefined();
+    expect(eslintConfig.overrides[0].rules['@typescript-eslint/no-deprecated']).toBe('warn');
```

Comment:
Hmm, would it make sense to test the whole content - just to make sure that it didn't touch anything else?
```ts
expect(eslintConfig).toEqual({
  overrides: [
    {
      files: ['src/**/*.{ts,tsx}'],
      rules: {
        '@typescript-eslint/no-deprecated': 'warn',
      },
    }
  ]
});
```